### PR TITLE
Gatekeeper Work

### PR DESCRIPTION
**Pull request by becky-intelletive**
TODO:

- Backend:
  - move remaining routes to Gatekeeper module
  - "add constraint" endpoint: add template validation
- Frontend GatekeeperService:
  - implement remaining routes
  - use new service
  - delete old service
  - delete old interfaces
- Frontend:
  - `add-template-constraint.component.ts` disregards non-form values from editing the raw manifest, effectively making the raw manifest editor useless

### CODE REVIEW

**Comment by becky-intelletive** [comment]
26 of these files are new 😮

File: dash/backend/src/exception-filters/http-exception.filter.ts [hunk 1]
```diff
@@ -17,13 +17,27 @@ export class HttpExceptionFilter implements ExceptionFilter {
...
     const response = ctx.getResponse<Response>();
-    
+
     // build response
     const status = exception.getStatus();
 
     let errorMessage = exception.message;
 
-    if (exception instanceof BadRequestException && typeof exception.getResponse === 'function' && 
+    const respFromException: any = exception.getResponse();
```
Comment (jacobbeasley) [inline dash/backend/src/exception-filters/http-exception.filter.ts:26]: Why was this changed?

Comment (becky-intelletive) [reply dash/backend/src/exception-filters/http-exception.filter.ts:26]: This filter stripped out any extra data provided by an error.

**Comment by becky-intelletive** [comment]
That's legacy behavior

File: dash/backend/src/exception-filters/http-exception.filter.ts (hunk 1 above)
Comment (becky-intelletive) [reply dash/backend/src/exception-filters/http-exception.filter.ts]: Recheck :)

**Review by jacobbeasley — APPROVED** [review]
Looks good!
